xtraplatform-feature-provider-sql/src/main/java/de/ii/xtraplatform/feature/provider/sql/app/ExtentReaderSql.java

```diff
@@ -61,4 +61,39 @@
         return LogContextStream.graphWithMdc(sourceStream.map(sqlRow -> sqlDialect.parseExtent((String) sqlRow.getValues()
                                                                                                               .get(0), crs)), Sink.head());
     }
+
+    public RunnableGraphWithMdc<CompletionStage<Optional<Interval>>> getTemporalExtent(FeatureStoreTypeInfo typeInfo, String property) {
+        FeatureStoreInstanceContainer instanceContainer = typeInfo.getInstanceContainers()
+                .get(0);
+        Optional<FeatureStoreAttributesContainer> temporalAttributesContainer = instanceContainer.getTemporalAttributesContainer(property);
+
+        if (!temporalAttributesContainer.isPresent()) {
+            throw new IllegalArgumentException("temporal property not found:" + property);
+        }
+
+        String query = queryGenerator.getTemporalExtentQuery(temporalAttributesContainer.get(), property);
+
+        Source<SqlRow, NotUsed> sourceStream = sqlConnector.getSqlClient().getSourceStream(query, SqlQueryOptions.withColumnTypes(String.class));
```

Comment:
```suggestion
        Source<SqlRow, NotUsed> sourceStream = sqlConnector.getSqlClient().getSourceStream(query, SqlQueryOptions.withColumnTypes(String.class, String.class));
```

You have to provide two column types because the query has two columns.